### PR TITLE
Add support for nested query in field params

### PR DESCRIPTION
**Pull request by tptrixtop**
Sorry, previous request was incorrect.

This one is correct.
Add support for multi limit in nested queries;
That fix need for nested queries inside field parameters;

In previous thread was mentioned that this is fix for that issue: #262 

Fix bug for nested queries inside field parameters;

### CODE REVIEW

**Comment by webimpress** [comment]
@tptrixtop as @froschdesign mentioned before in https://github.com/zendframework/zend-db/pull/299#issuecomment-358620450 can you add test case for it, please?

**Comment by tptrixtop** [comment]
I have add tests for that case. 
And 1 more fix for the same issue with offset param in nested queries;

File: src/Sql/Platform/Mysql/SelectDecorator.php [hunk 1]
```diff
@@ -50,7 +50,8 @@ protected function processLimit(
             return;
         }
         if ($parameterContainer) {
-            $parameterContainer->offsetSet('limit', $this->limit, ParameterContainer::TYPE_INTEGER);
+            $param_prefix = $this->processInfo['paramPrefix'];
```
Comment (webimpress) [inline src/Sql/Platform/Mysql/SelectDecorator.php]: Please use camelCase variable names instead of underscore_separated style.

Comment (tptrixtop) [reply src/Sql/Platform/Mysql/SelectDecorator.php]: Yes, i think case with 1 query is valid, i will add it)

**Comment by tptrixtop** [comment]
Done. I have added test for single nested query ( few limit case #262 ).
Variable renamed from underscore to camel case.

File: test/Sql/Platform/Mysql/SelectDecoratorTest.php [hunk 2]
```diff
@@ -95,10 +96,74 @@ public function dataProvider()
...
+            ->columns([
+                'cnt' => new Expression('count(foo1.id)')
+            ])->limit(0)->offset(5);
+
+        $select3 = new Select;
+        $select3->from('foo')
+            ->columns([
+                'res'  => $nestedSelect0,
+            ])
+            ->limit('10')->offset('5');
```
Comment (webimpress) [inline test/Sql/Platform/Mysql/SelectDecoratorTest.php]: I would set here different offset, now we have the same as in the nested query, and it's possible that something is getting mixed, so to make it clear I would change the value.

Comment (webimpress) [inline test/Sql/Platform/Mysql/SelectDecoratorTest.php]: Another question also here: why these values are passed as strings, not as integers?

File: test/Sql/Platform/Mysql/SelectDecoratorTest.php [hunk 3]
```diff
@@ -95,10 +96,74 @@ public function dataProvider()
...
+                'cnt' => new Expression('count(foo2.id)')
+            ])->limit(50)->offset(101);
+
+        $select4 = new Select;
+        $select4->from('foo')
+            ->columns([
+                'res'  => $nestedSelect0,
+                'res0' => $nestedSelect1
+            ])
+            ->limit('10')->offset('5');
```
Comment (webimpress) [inline test/Sql/Platform/Mysql/SelectDecoratorTest.php]: Also here, I would change the offset, to be different than in nested queries.

**Comment by tptrixtop** [comment]
I have just copied them from my working code)

**Comment by tptrixtop** [comment]
I have remove extra quotes and change offset & limit.

I did not check it for other platforms. Only for MySQL. 
I will try to check it for other platforms)

**Comment by tptrixtop** [comment]
Done.
Limit & Offset are working in Oracle, MS Sql & IbmDb2.

Only additional fix needed in Sqlite. I have fixed it & added test for that & committed;

File: src/Sql/Platform/Sqlite/SelectDecorator.php [hunk 4]
```diff
@@ -55,6 +55,43 @@ protected function processStatementStart(
...
+            return;
+        }
+        if ($parameterContainer) {
+            $paramPrefix = $this->processInfo['paramPrefix'];
+            $parameterContainer->offsetSet($paramPrefix . 'offset', $this->offset, ParameterContainer::TYPE_INTEGER);
+            return [$driver->formatParameterName('offset')];
+        }
+
+        return [$this->offset];
+    }
```
Comment (webimpress) [inline src/Sql/Platform/Sqlite/SelectDecorator.php]: I'd just to note that these methods are exactly the same as in:
https://github.com/zendframework/zend-db/blob/4675b97a5991279313818145538b70558419e410/src/Sql/Platform/Mysql/SelectDecorator.php#L41-L76

and the difference with parent class `Select` is to not quote offset and limit values.
I think it's fine, for MySQL it was added in that commit: https://github.com/zendframework/zend-db/commit/91bfba7944e36702c79e9acba7ebd7022b9a4611

**Comment by ezimuel** [comment]
@tptrixtop thanks!

**Pull request merged**